Source/Particles/Pusher/GetAndSetPosition.H

```diff
@@ -0,0 +1,76 @@
+#ifndef WARPX_PARTICLES_PUSHER_GETANDSETPOSITION_H_
+#define WARPX_PARTICLES_PUSHER_GETANDSETPOSITION_H_
+
+#include <limits>
+#include <WarpXParticleContainer.H>
+#include <AMReX_REAL.H>
+
+#ifndef WARPX_RZ
+
+/* \brief Extract the particle's coordinates from the ParticleType struct `p`,
+ *        and stores them in the variables `x`, `y`, `z`. */
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void GetPosition(
```

Comment:
That is probably a stupid question, but could we make this function (and others) a member of `WarpXParticleContainer`? Would it work with inlining on GPU? That could help having something more general (i.e. avoid duplication for Boris/Vay pusher, charged particles/photons).

Reply:
I am not familiar with Boris/Vay pusher and charged particles/photons.  Could you elaborate on why being a member function could make them more general?

That being said, I don't think they can be a member function (unless it does not use any data in the object and therefore doesn't need to be a member function).   If the function needs to use any data in `WarpXParticleContainer`, it won't work on gpu because the particle container object is not in gpu memory.

Reply:
unless we are talking about static constexpr data.

Reply:
Boris and Vay pushers are very similar, as an illustration the velocity update is shown on the image below, for each of them.
<img width="605" alt="Screen Shot 2019-05-22 at 1 00 12 PM" src="https://user-images.githubusercontent.com/26292713/58204721-bdec6880-7c91-11e9-9378-cb2e3e1777af.png">
So maybe we could do something like
```
amrex::ParallelFor(
   // inline my_pusher.push(x,y,z,<etc.>)
);
```
where `my_pusher` is an instance of class `BorisPusher` or `VayPusher`, for instance (without needing member data). Again, I would have to check.

As far as photons are concerned, the push is quite different: photons do not have a `gamma`, they just go straight at the speed of light. The `UpdatePosition` would do 
```
x += vx * dt;
```
instead of
```
x += ux * inv_gamma * dt;
```
where `vx` is the `x` component of a 3D vector which norm equals `c`.
The `UpdateMomentum` would do nothing, I guess...